spec/syntax.md

```diff
@@ -457,6 +465,10 @@ Otherwise, the set of characters allowed in names is large.
 The _nmtoken_ token doesn't have _name_'s restriction on the first character
 and is used as variant keys and option values.
 
+The _argtoken_ is used for expression arguments,
+and matches _nmtoken_ except for not allowing `-` or `:` as a first character.
+It allows some literal argument values to be unquoted.
```

Comment:
This is the part of this PR that keeps bringing me back for another look. I don't know what an "expression argument" is, especially given that other tokens found inside an expression can be considered arguments (notably, `options` can be considered arguments to a `function` or a `reserved`).

Since the purpose of `argtoken` is to allow unquoted literals, I think I might prefer if we dealt with them by recasting `literal` and naming `argtoken` as `unquoted`, e.g.:

```abnf
; literal data values; some values, depending on content, can appear
; without quoting | delimiters
literal  = unquoted / ("|" *(literal-char / literal-escape) "|")
unquoted = arg-start *name-char
```

Otherwise I'd want to rewrite aspects of this text to clarify "expression argument"

Reply:
I've now renamed the added tokens as `unquoted` and `unquoted-start`.